org.eclipse.winery.repository.rest/src/main/java/org/eclipse/winery/repository/rest/edmm/EdmmResource.java

```diff
@@ -0,0 +1,126 @@
+package org.eclipse.winery.repository.rest.edmm;
+
+import java.io.IOException;
+import java.util.List;
+import java.io.File;
+import java.io.StringWriter;
+import java.util.UUID;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import io.github.edmm.core.parser.EntityGraph;
+import io.github.edmm.core.plugin.PluginService;
+import io.github.edmm.model.DeploymentModel;
+import io.github.edmm.model.PluginSupportResult;
+import org.apache.commons.io.FileUtils;
+
+import org.eclipse.winery.common.configuration.Environments;
+import org.eclipse.winery.common.constants.MimeTypes;
+import org.eclipse.winery.edmm.TransformationManager;
+import org.eclipse.winery.edmm.plugins.PluginManager;
+import org.eclipse.winery.model.tosca.TExtensibleElements;
+import org.eclipse.winery.model.tosca.TServiceTemplate;
+import org.eclipse.winery.repository.rest.RestUtils;
+
+/**
+ * This class manages the API for the EDMM functionalities
+ * like the Decision Support System and
+ * the Transformation Framework
+ */
+public class EdmmResource {
+
+    private TExtensibleElements element = null;
+    
+    public EdmmResource(TExtensibleElements element) {
+        this.element = element;
+    }
+
+    /**
+     * Winery at the moment doesn't create a valid EntityGraph, but a graph that will become valid only after 
+     * the conversion to a YAML file. It works like this because originally only the YAML string was needed.
+     * 
+     * Now we use a function that takes the 'invalid' EntityGraph converts it to a valid YAML and then gets back
+     * a valid EntityGraph.
+     * 
+     */
+    private EntityGraph correctGraph(EntityGraph graph) {
```

Comment:
Shouldn't be part of this PR.

Reply:
But without that correction it doesn't work. Where should I put that? In the edmm-core?

Reply:
Ah, misunderstood the method. It's fine